From c5d6b02d7ed141a4dae7648dd65a09100db49f97 Mon Sep 17 00:00:00 2001
From: Lawrence Rust <lvr@softsystem.co.uk>
Date: Fri, 16 Aug 2013 14:08:02 +0100
Subject: [PATCH 203/227] myth_system: Don't redirect stdout/err to dev/null

This patch reinstates stdout/stderr output for programs run with
myth_system.

At present it's very difficult to see what is happening and fix any
errors in user provided scripts and/or programs.

Signed-off-by: Lawrence Rust <lvr@softsystem.co.uk>
---
 mythtv/libs/libmythbase/system-unix.cpp |    6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/mythtv/libs/libmythbase/system-unix.cpp b/mythtv/libs/libmythbase/system-unix.cpp
index 2c508d3..d033d29 100644
--- a/mythtv/libs/libmythbase/system-unix.cpp
+++ b/mythtv/libs/libmythbase/system-unix.cpp
@@ -928,6 +928,7 @@ void MythSystemUnix::Fork(time_t timeout)
                 _exit(GENERIC_EXIT_PIPE_FAILURE);
             }
         }
+#if 0
         else
         {
             /* We aren't sucking this down, redirect stdout to /dev/null */
@@ -950,7 +951,7 @@ void MythSystemUnix::Fork(time_t timeout)
                      << strerror(errno) << endl;
             }
         }
-
+#endif
         /* handle standard err */
         if( p_stderr[1] >= 0 )
         {
@@ -963,6 +964,7 @@ void MythSystemUnix::Fork(time_t timeout)
                 _exit(GENERIC_EXIT_PIPE_FAILURE);
             }
         }
+#if 0
         else
         {
             /* We aren't sucking this down, redirect stderr to /dev/null */
@@ -985,7 +987,7 @@ void MythSystemUnix::Fork(time_t timeout)
                      << strerror(errno) << endl;
             }
         }
-
+#endif
         /* Close all open file descriptors except stdin/stdout/stderr */
         for( int i = sysconf(_SC_OPEN_MAX) - 1; i > 2; i-- )
             close(i);
-- 
1.7.9.5

